pkgs/applications/networking/browsers/firedragon/default.nix

```
@@ -0,0 +1,115 @@
+{
+  buildMozillaMach,
+  callPackage,
+  fetchFromGitLab,
+  fetchurl,
+  lib,
+  nixosTests,
+  pkgs,
+  python311,
+  stdenv,
+}:
+let
+  current = lib.trivial.importJSON ./version.json;
+  packageVersion = current.firedragonSource.version;
+  version = current.firefoxVersion;
+
+  settings = fetchFromGitLab {
+    owner = "garuda-linux/firedragon";
+    repo = "settings";
+    fetchSubmodules = false;
+    inherit (current.firedragonSettings) rev hash;
+  };
+in
+(
+  (buildMozillaMach rec {
+    pname = "firedragon";
+    applicationName = "FireDragon";
+    binaryName = "firedragon";
+    branding = "browser/branding/firedragon";
+    requireSigning = false;
+    allowAddonSideload = true;
+
+    inherit packageVersion version;
+
+    src = fetchurl {
+      url = "https://gitlab.com/api/v4/projects/55893651/packages/generic/firedragon/${packageVersion}/firedragon-v${packageVersion}.source.tar.zst";
+      inherit (current.firedragonSource) hash;
+    };
+
+    updateScript = callPackage ./update.nix { };
+
+    tests = {
+      inherit (nixosTests) firedragon;
+    };
+
+    extraConfigureFlags = [
+      "--disable-crashreporter"
+      "--disable-debug"
+      "--disable-debug-js-modules"
+      "--disable-debug-symbols"
+      "--disable-default-browser-agent"
+      "--disable-gpsd"
+      "--disable-necko-wifi"
+      "--disable-rust-tests"
+      "--disable-tests"
+      "--disable-updater"
+      "--disable-warnings-as-errors"
+      "--disable-webspeech"
+      "--enable-bundled-fonts"
+      "--enable-jxl"
+      "--enable-private-components"
+      "--enable-proxy-bypass-protection"
+      "--with-app-name=${pname}"
+      "--with-app-basename=${applicationName}"
+      "--with-distribution-id=org.garudalinux"
+      "--with-unsigned-addon-scopes=app,system"
+    ];
```

The following are already handled in `buildMozillaMach`:
- `debug` is already handled by the `debugBuild` flag `false` by default
- `necko-wifi` is already handled by the `crashreporterSupport` option;
- `crashreporter` is handled by `geolocationSupport` which is false when `privacySupport` is true;
- `tests` and `updater` are currently always disabled in `buildMozillaMach`
- `distribution-id` get set to `org.nixos` in `buildMozillaMach` and should stay like that even in forks of Firefox, as it identifies to the users that this build was distributed through nixpkgs.

```suggestion
    extraConfigureFlags = [
      "--disable-debug-js-modules"
      "--disable-debug-symbols"
      "--disable-default-browser-agent"
      "--disable-gpsd"
      "--disable-rust-tests"
      "--disable-warnings-as-errors"
      "--disable-webspeech"
      "--enable-bundled-fonts"
      "--enable-jxl"
      "--enable-private-components"
      "--enable-proxy-bypass-protection"
      "--with-app-name=${pname}"
      "--with-app-basename=${applicationName}"
      "--with-unsigned-addon-scopes=app,system"
    ];
```

---

Thanks, fixed! :)